design/frontend-design.md

```diff
@@ -0,0 +1,147 @@
+## Test Cases
+1.  class with two fields
+```
+    input:
+        class C:
+            a : int = 1
+            b : int = 2
+        c : C = None
+        c = C()
+    // in REPL:
+    >>> c
+    
+
+    output:
+        {
+            address: 100
+            a: 1
+            b: 2
+        }
+```
+
+2. class with nested class instance
+```
+   input:
+       class E(object):
+            a : int = 1
+       class C(object):
+            a : bool = True
+            e : E = None
+            def __init__(self: C):
+               self.e = E()
+            def d(self: C) -> int:
+              return 1
+        c : C = None
+        c = C()
+    // in REPL:
+    >>> c
+        
+    output:
+        {
+            address: 100
+            a: True
+            e: {
+                address: 108
+                a: 1
+            }
+        }
+```
+
+3.  class with inheritance: 
+ we need to wait for the inheritance group to finish their feature.
+```
+input: 
+    class E(object):
+        a : int = 1
+        f : int = 2
+    class C(E):
+        a : int = 2
+        e : E = None
+        def __init__(self: C):
+           self.e = E()
+        def d(self: C) -> int:
+          return 1
+    c : C = None
+    c = C()
+    // in REPL:
+    >>> c
+
+output: 
+      {
+            address: 100
+            a: 2,
+            f: 2,
+            e: {
+                address: 112
+                a: 1
+            }
+        }
+```
+
+4. nested instance with uninitilized instace
+```
+    input:
+    class E(object):
+        a : int = 1
+
+    class C(E):
+        a : int = 2
+        e : E = None
+        def d(self: C) -> int:
+            return 1
+    c : C = None
+    c = C()
+    // in REPL:
+    >>> c
+
+    output:
+        {
+            address: 0x1234
+            a: 1
+            e: none
+        }
+```
+5.  Add one HTML component <select></select>:
+It would be much more convenient for users to begin with some sample code. We could provide a HTML component listing the example codes. For example, the options of the select could be "example code with class", "example code with closure". Once we click one option, the code would be loaded to the left box on the screen.
+
+6.  Switch the select component:
+When we change the option of the select, the example code would be overwritten. When we clicked the "example code with closure" option, the code in the left box would be overwritten.
+
+7. "Clear" button:
```

Comment:
Can we put the buttons in better places, too? Like think about nice alignment for them?